src/partitionAccessory.ts

```diff
@@ -137,10 +138,21 @@ export class EnvisalinkPartitionAccessory {
             case 'entrydelay':
             case 'arming':
             case 'busy':
+            case 'troubleledoff':
                 break;
-            default:
+            case 'troubleledon':
+                this.platform.log.info(`Ignoring status '${this.partition.status.text}' (${this.partition.status.description})`);
+                break;
+            case 'ready':
+            case 'readyforce':
                 currentState = this.platform.Characteristic.SecuritySystemCurrentState.DISARMED;
                 targetState = this.platform.Characteristic.SecuritySystemTargetState.DISARM;
+                break;
+            default:
+                // If we do not recognise the security pannel state then we should not assume
+                // that statis is DISARMED.  Leave it unchanged and log warning message.  With
+                // luck user will report this and provide debug trace.
+                this.platform.log.warn(`Ignoring status '${this.partition.status.text}' (${this.partition.status.description})${this.platform.reportError}`);
```

Comment:
I am fairly certain this is not the behavior we want. There is a slew of statuses that will need to be handled. You can find all of them here: https://github.com/dustindclark/NodeAlarmProxy/blob/49d817d3503740911eb54ac8ccade817a56554b8/envisalink.js

This has caused issues in the past iirc.

Reply:
I disagree.  If there are messages that are known to affect alarm state (require reporting a state change to STAY_ARM, AWAY_ARM, NIGHT_ARM, DISARMED, ALARM_TRIGGERED) then they should be explicitly handled in the code and not allowed to fall through to some default.  That was the original bug that prompted issue #200 and this PR.

There are 92 entries in that file of which 21 are `applicationcommands` and 71 are `tpicommands`.  It looks like we only need to handle the `tpicommands`?  Of those 56 have message text of which the plugin currently explicitly handles 19 (two of which are not in the node alarm file).  Leaving 39 that fall through to `default`.  Should all of these result in setting the alarm state to DISARMED?  I don't think so, but that is what the code was doing.

Correct way to handle this is to identify and explicitly handle those messages that should change the state.

Currently handled by the plugin...
```
alarm          // sets ALARM_TRIGGERED
specialclosing // ignored with debug message
partialclosing // ignored with debug message
chimeenabled   // ignored with debug message
chimedisabled  // ignored with debug message
armed          // sets STAY_ALARM, NIGHT_ALARM or AWAY_ALARM
armedbypass    // not in the file, same handling as armed
disarmed       // sets DISARMED
notready       // sets DISARMED
failedtoarm    // sets DISARMED
useropening    // sets DISARMED
exitdelay      // ignored with no messages
entrydelay     // ignored with no messages
arming         // not in file, ignored with no messages
busy           // ignored with no messages
troubleledoff  // ignored with no messages
troubleledon   // ignored with info message
ready          // set DISARMED
readyforce     // set DISARMED
```

Not handled by the plugin...
```
alarmrestore  
tamper
tamperrestore
fault
faultrestore
open
openrestore
duress
firekeyalarm
firekeyrestore
auxkeyalarm
auxkeyrestore
panickeyalarm
panickeyrestore
twowiresmokealarm
twowiresmokerestore
keypadlockout
failedtoarm
pgmoutputinprogress
invalidcode
functionnotavailable
autoarming
installersmode
userclosing
specialopening
batterytrouble
batterytroublerestore
actrouble
actroublerestore
systembelltrouble
systembelltroublerestore
communicationtrouble
buffernearfull
tamper
tamperrestore
firetroublealarm
firetroublealarmrestore
verbosetroublestatus
```
If any of these should set state then they should be explicitly handled in the code.